artemis-protocols/artemis-openwire-protocol/src/test/java/org/apache/activemq/artemis/core/protocol/openwire/amq/OpenWireProtocolManagerTest.java

```diff
@@ -0,0 +1,78 @@
+import org.apache.activemq.artemis.utils.RandomUtil;
+import org.apache.activemq.artemis.utils.actors.ArtemisExecutor;
+import org.apache.activemq.command.ConnectionId;
+import org.apache.activemq.command.ConnectionInfo;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class OpenWireProtocolManagerTest {
+
+   @Test
+   public void testNullPrimaryOnNodeUp() throws Exception {
+
+      ArtemisExecutor executor = ArtemisExecutor.delegate(Executors.newSingleThreadExecutor(ActiveMQThreadFactory.defaultThreadFactory(getClass().getName())));
+
+      ClusterManager clusterManager = Mockito.mock(ClusterManager.class);
+      ActiveMQServer server = Mockito.mock(ActiveMQServer.class);
+      StorageManager storageManager = new NullStorageManager();
+      Mockito.when(server.getStorageManager()).thenReturn(storageManager);
+      Mockito.when(server.newOperationContext()).thenReturn(storageManager.newContext(executor));
+      Mockito.when(server.getClusterManager()).thenReturn(clusterManager);
+      Mockito.when(clusterManager.getDefaultConnection(Mockito.any())).thenReturn(null);
+      SecurityStore securityStore = Mockito.mock(SecurityStore.class);
+      Mockito.when(server.getSecurityStore()).thenReturn(securityStore);
+      Mockito.when(securityStore.authenticate(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(null);
+      ServerSession serverSession = Mockito.mock(ServerSession.class);
+      Mockito.when(serverSession.getName()).thenReturn("session");
+      Mockito.doReturn(serverSession).when(server).createSession(Mockito.anyString(), Mockito.any(), Mockito.any(), Mockito.anyInt(), Mockito.any(), Mockito.anyBoolean(), Mockito.anyBoolean(), Mockito.anyBoolean(),
+                                                                 Mockito.anyBoolean(), Mockito.any(), Mockito.any(), Mockito.anyBoolean(), Mockito.any(), Mockito.any(), Mockito.anyString(), Mockito.any(), Mockito.anyBoolean());
+
+      OpenWireProtocolManager openWireProtocolManager = new OpenWireProtocolManager(null, server,null, null);
+      openWireProtocolManager.setSecurityDomain("securityDomain");
+      openWireProtocolManager.setSupportAdvisory(false);
+      Connection connection = Mockito.mock(Connection.class);
+      Mockito.doReturn(new ChannelBufferWrapper(Unpooled.buffer(1024))).when(connection).createTransportBuffer(Mockito.anyInt());
+      OpenWireConnection openWireConnection = new OpenWireConnection(connection, server, openWireProtocolManager, openWireProtocolManager.wireFormat(), executor);
+      ConnectionInfo connectionInfo = new ConnectionInfo(new ConnectionId("1:1"));
+      connectionInfo.setClientId(RandomUtil.randomString());
+      openWireProtocolManager.addConnection(openWireConnection, connectionInfo);
+
+      TopologyMember topologyMember = new TopologyMemberImpl(RandomUtil.randomString(), null, null, null, null);
+      openWireProtocolManager.nodeUP(topologyMember, false);
+      executor.shutdown();
```

Comment:
This still wont happen if something pops earlier. Needs at least a try-finally. Or you could use e.g ArtemisTestCase as a parent and use the runAfter[Ex] method as you creating the executor.